### PR TITLE
Allow xhtml endpoints

### PR DESCRIPTION
**Pull request by Henry-Pulver**
Also added basic tests for forming HTTP requests

### CODE REVIEW

File: lib/edge-runtime/requests.ts [hunk 1]
```diff
@@ -149,40 +172,62 @@ export async function httpRequestFromAction({
...
+  // You'll get it!
+  return { url, requestOptions };
+  // The Will to Win (Berton Braley)
+}
+
+export async function makeHttpRequest(
+  url: string,
+  requestOptions: RequestInit
+): Promise<any> {
   // Deal with response with potentially empty body (stackoverflow.com/a/51320025)
```
Comment (neuromantic99) [inline lib/edge-runtime/requests.ts:199]: are you still dealing with this?

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "@uppy/react": "^3.1.2",
     "@upstash/ratelimit": "^0.4.3",
     "@upstash/redis": "^1.22.0",
+    "cheerio": "^1.0.0-rc.12",
```
Comment (neuromantic99) [inline package.json:32]: 👋

File: pages/api/parse-html.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
Comment (neuromantic99) [inline pages/api/parse-html.ts:1]: what exactly is the reason for making this an endpoint rather than a function?

File: pages/api/parse-html.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    .replaceAll(/\n{2,}/g, "\n\n")
+    .trim();
+
+  res.status(200).send(text);
+}
+
+function removeHiddenElements(
+  body: cheerio.Cheerio<cheerio.Element>
+): cheerio.Cheerio<cheerio.Element> {
+  // Absurd, but can't find a better way to remove hidden elements
```
Comment (neuromantic99) [inline pages/api/parse-html.ts:47]: 😆

File: pages/api/parse-html.ts (hunk 3 above)
Comment (Henry-Pulver) [reply pages/api/parse-html.ts:1]: It runs on the edge runtime. Edge functions can't import arbitrary node packages

File: pages/api/parse-html.ts (hunk 4 above)
Comment (Henry-Pulver) [reply pages/api/parse-html.ts:47]: querySelectorAll doesn't exist :cry:

Comment (Henry-Pulver) [reply pages/api/parse-html.ts:47]: Found a better alternative

**Pull request merged**